content/en/docs/tasks/observability/metrics/tcp-metrics/index.md

```diff
@@ -0,0 +1,160 @@
+application.
+
+* This task assumes that the Bookinfo sample will be deployed in the `default`
+namespace. If you use a different namespace, update the
+example configuration and commands.
+
+## Collecting new telemetry data
+
+1.  Setup Bookinfo to use MongoDB.
+
+    1.  Install `v2` of the `ratings` service.
+
+        If you are using a cluster with automatic sidecar injection enabled,
+        deploy the services using `kubectl`:
+
+        {{< text bash >}}
+        $ kubectl apply -f @samples/bookinfo/platform/kube/bookinfo-ratings-v2.yaml@
+        {{< /text >}}
+
+        If you are using manual sidecar injection, run the following command instead:
+
+        {{< text bash >}}
+        $ kubectl apply -f <(istioctl kube-inject -f @samples/bookinfo/platform/kube/bookinfo-ratings-v2.yaml@)
+        deployment "ratings-v2" configured
+        {{< /text >}}
+
+    1.  Install the `mongodb` service:
+
+        If you are using a cluster with automatic sidecar injection enabled,
+        deploy the services using `kubectl`:
+
+        {{< text bash >}}
+        $ kubectl apply -f @samples/bookinfo/platform/kube/bookinfo-db.yaml@
+        {{< /text >}}
+
+        If you are using manual sidecar injection, run the following command instead:
+
+        {{< text bash >}}
+        $ kubectl apply -f <(istioctl kube-inject -f @samples/bookinfo/platform/kube/bookinfo-db.yaml@)
+        service "mongodb" configured
+        deployment "mongodb-v1" configured
+        {{< /text >}}
+
+    1.  The Bookinfo sample deploys multiple versions of each microservice, so begin by creating destination rules
+        that define the service subsets corresponding to each version, and the load balancing policy for each subset.
+
+        {{< text bash >}}
+        $ kubectl apply -f @samples/bookinfo/networking/destination-rule-all.yaml@
+        {{< /text >}}
+
+        If you enabled mutual TLS, run the following command instead:
+
+        {{< text bash >}}
+        $ kubectl apply -f @samples/bookinfo/networking/destination-rule-all-mtls.yaml@
+        {{< /text >}}
+
+        To display the destination rules, run the following command:
+
+        {{< text bash >}}
+        $ kubectl get destinationrules -o yaml
+        {{< /text >}}
+
+        Wait a few seconds for destination rules to propagate before adding virtual services that refer to these subsets, because the subset references in virtual services rely on the destination rules.
+
+    1.  Create `ratings` and `reviews` virtual services:
+
+        {{< text bash >}}
+        $ kubectl apply -f @samples/bookinfo/networking/virtual-service-ratings-db.yaml@
+        Created config virtual-service/default/reviews at revision 3003
+        Created config virtual-service/default/ratings at revision 3004
+        {{< /text >}}
+
+1.  Send traffic to the sample application.
+
+    For the Bookinfo sample, visit `http://$GATEWAY_URL/productpage` in your web
+    browser or use the following command:
```

Comment:
I'm not sure how far down the `Deploy Bookinfo` the user would have gotten in the prior step. GATEWAY_URL isn't set until the `Confirm the app is accessible from outside the cluster` step. The step after that is applying the `destination-rule-all` yaml which you do call out here. Maybe @adammil2000 or @frankbu have more experience on what it means to say `Deploy Bookinfo` and how far the user may have gone.

Reply:
We did similar thing for previous tcp metrics task too and don't think there was a problem... https://preliminary.istio.io/docs/tasks/observability/mixer/metrics/tcp-metrics/

Reply:
Eric is correct, you're missing the steps where a value actually gets assigned to GATEWAY_URL to use it here. Check out the steps we used here to make that happen correctly: https://preliminary.istio.io/docs/setup/getting-started/#determining-the-ingress-ip-and-ports

Reply:
Thanks.. Fixed